docs/DebuggingNumericalError.md

```diff
@@ -100,3 +100,29 @@ To print a message followed by one value, inject the following code (where `val`
 ```code
 create.krnl.printf("inputElem: ", val, valType);
 ```
+
+## Finding memory errors
+
+If you know, or suspect, that onnx-mlir-compiled inference executable causes
+bugs related to memory allcations. The valgrind
+framework(https://valgrind.org/) or mtrace memory
+tool(https://github.com/sstefani/mtrace) are available. These tools trace memory
+allocation/free-related APIs, and can fix memory issues, such as memory leaks.
+
+However if the bugs relating to memory access, especially buffer overrun bugs,
+it is difficult to debug because run-time errors occur outside of the point
+ containing the bug. 
+The "Electric Fence library"(https://github.com/CheggEng/electric-fence) can be
+used for debugging these bugs. It helps you detect two common programming bugs: software that overruns the boundaries of a malloc() memory allocation, and
+software that touches a memory allocation
+that has been released by free(). Unlike other malloc() debuggers, Electric
```

Comment:
"malloc() debuggers" --> memory debuggers ?

Reply:
Done. Thank you.